### PR TITLE
feat: Report with Dynamic data

### PR DESCRIPTION
**Pull request by vaibhavsingh97**

### CODE REVIEW

File: src/components/EnvironmentDropdown/index.tsx [hunk 1]
```diff
@@ -11,24 +10,14 @@ import {
...
-};
+import { EnvironmentData, EnvironmentDropdownProps } from './types';
+import { filterData } from './utils';
 
 const EnvironmentDropdown: React.FC<EnvironmentDropdownProps> = ({ data }) => {
+  const [query, setQuery] = useState<string>('');
+  const filteredData = filterData(data, query);
   const [isDropDownOpen, setDropdownOpen] = useState(false);
-  const [context, setContext] = useState('Environment 2');
+  const [context, setContext] = useState(filteredData[0].name);
```
Comment (vaibhavsingh97) [inline src/components/EnvironmentDropdown/index.tsx:20]: filteredData contains an array of objects with the highest number of failures, so I am updating the state using the index. Can this be optimized?

File: src/components/EnvironmentDropdown/index.tsx [hunk 2]
```diff
@@ -33,9 +33,9 @@ const EnvironmentDropdown: React.FC<EnvironmentDropdownProps> = ({ data }) => {
       <DropdownMenuTrigger asChild>
         <EnvironmentSelectorWrapper>
           <EnvironmentContent
-            name={envData.name || data[0]['name']}
-            meta={envData.meta || data[0]['meta']}
-            testResult={envData.testResult || data[0]['testResult']}
+            name={envData.name || filteredData[0]['name']}
```
Comment (vaibhavsingh97) [inline src/components/EnvironmentDropdown/index.tsx:36]: Is there any better approach to creating components with default data?

File: src/types/nightwatch.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
+export {};
```
Comment (SwarajGK) [inline src/types/nightwatch.ts:1]: why?

File: src/transform.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+    envData[envName]['metadata'] = environmentData.metadata;
+    envData[envName]['stats'] = environmentData.stats;
+
+    Object.keys(environmentDataFiles).forEach((fileName) => {
+      const fileData = {} as IFileStats;
+      const fileReport = environmentDataFiles[fileName];
+      const metadata = envData[envName].metadata;
+
+      fileData['fileName'] = fileName;
```
Comment (SwarajGK) [inline src/transform.ts:62]: `fileData.fileName = fileName;`

Same can be done for all keys

File: src/helper/findFailedTestDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+interface IFailedData {
+  name: string;
+  fileIndex: string;
+  testIndex: string;
+}
+
+const findFailedTests = (file: IFileStats) => {
+  const test = file.tests.find((test) => {
+    return test.status === file.status && test.key;
+  });
```
Comment (SwarajGK) [inline src/helper/findFailedTestDetails.tsx:42]: `const test = file.tests.find((test) => test.status === file.status && test.key);`

File: package.json [hunk 6]
```diff
@@ -2,33 +2,27 @@
...
+    "build": "tsc && vite build && tsc -p tsconfig.node.cjs.json",
     "preview": "vite preview",
     "lint": "eslint .",
     "lint:fix": "eslint --fix",
-    "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc"
+    "format": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}' --config ./.prettierrc",
+    "clean": "rm -rf dist index.js index.d.ts *.tsbuildinfo *.tgz"
   },
-  "dependencies": {
+  "devDependencies": {
```
Comment (SwarajGK) [inline package.json:15]: why dev `@emotion/react` and other UI lib are part of `devDependencies` and not `dependencies`

File: src/components/TestDetailsView/utils.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    return false;
+  }
+
+  if (args && args.length < 1) {
+    return false;
+  }
+
+  if (args && args[0] === null) {
+    return false;
+  }
```
Comment (SwarajGK) [inline src/components/TestDetailsView/utils.ts:32]: can combine everything 
`return args && args.length > 1 && args[0] !== null`

**Review by SwarajGK — APPROVED** [review]
Approving with comments

**Pull request merged**